src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -1690,18 +1694,24 @@ BufferSupplier getSSLBufferSupplier() {
         return sslBufferSupplier;
     }
 
-    private Duration getIdleConnectionTimeoutProp() {
-        String s = Utils.getNetProperty("jdk.httpclient.keepalive.timeout");
-        if (s != null) {
-            try {
-                long val = Long.parseLong(s);
-                if (val >= 0)
-                    return Duration.ofSeconds(Long.parseLong(s));
-            } catch (NumberFormatException ignored) {}
-        }
+    private Duration getIdleConnectionTimeout() {
+        if (IDLE_CONNECTION_TIMEOUT >= 0)
+            return Duration.ofSeconds(IDLE_CONNECTION_TIMEOUT);
         return null;
     }
 
+    private static long getTimeoutProp(String prop, long def) {
+        String s = Utils.getNetProperty(prop);
+        try {
+            if (s != null) {
+                long timeoutVal = Long.parseLong(s);
+                System.err.println(timeoutVal);
+                if (timeoutVal >= 0) return timeoutVal;
+            }
+        } catch (NumberFormatException ignored) {}
```

Comment:
Similarly - you could use `Log.logTrace` to print the error. It could be useful for diagnostic.

Reply:
Good spot, would be useful. Should result in better logging with the test too. I'll include this.